d2bs/kolbot/libs/common/Attack.js

```diff
@@ -181,7 +222,7 @@ var Attack = {
 			}
 
 			if (Config.MFSwitchPercent && target.hp / 128 * 100 < Config.MFSwitchPercent) {
-				Precast.weaponSwitch(Math.abs(Config.MFSwitch));
+				Precast.weaponSwitch(Attack.getPrimarySlot() ^ 1);
```

Comment:
move precast.weaponswitch to attack.js, I don't see anything in that func that is precast specific